test/testdata/rbi/array.rb

```diff
@@ -39,6 +39,12 @@
 T.assert_type!([1,2].combination(1) {}, T::Array[Integer])
 T.assert_type!([1,2].repeated_combination(1) {}, T::Array[Integer])
 
+# assignments
+arr = [1, 2, 3]
+T.assert_type!(arr[0] = 1, Integer)
+T.assert_type!(arr[1..2] = 3, Integer)
+T.assert_type!(arr[1..2] = [100, 200], T::Array[Integer])
```

Comment:
We still have some tests that use `T.assert_type!` but we generally don't like using them in new tests because their meaning can subtly change over time. e.g., asserting that something is a `T::Enumerable[Integer]` when in fact a change has caused it to become a `T::Array[Integer]` silently. `T.reveal_type` would capture that change, but `T.assert_type` would not.

No change needed for this PR, especially because you were just copying what you saw. I just wanted to give you a heads up that one day I would love to not have `T.assert_type` in any tests.

Reply:
Interesting! Will follow up with some cleanup around here... It's also probably worth adding some documentation on how to test `rbi` changes :)